text_generation/causal_lm/cpp/beam_search_causal_lm.cpp

```diff
@@ -0,0 +1,84 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#include <group_beam_searcher.hpp>
+#include <openvino/openvino.hpp>
+#include <openvino_extensions/strings.hpp>
+
+namespace {
+std::pair<ov::Tensor, ov::Tensor> tokenize(ov::InferRequest&& tokenizer, std::string_view prompt) {
+    ov::Tensor destination = tokenizer.get_input_tensor();
+    openvino_extensions::pack_strings(std::array{prompt}, destination);
+    tokenizer.infer();
+    return {tokenizer.get_tensor("input_ids"), tokenizer.get_tensor("attention_mask")};
+}
+
+std::string detokenize(ov::InferRequest& detokenizer, const std::vector<int64_t>& tokens) {
+    constexpr size_t BATCH_SIZE = 1;
+    ov::Tensor inp = detokenizer.get_input_tensor();
+    inp.set_shape({BATCH_SIZE, tokens.size()});
+    for (size_t idx = 0; idx < tokens.size(); ++idx) {
+        inp.data<int64_t>()[idx] = tokens.at(idx);
+    }
+    detokenizer.infer();
+    return openvino_extensions::unpack_strings(detokenizer.get_output_tensor()).front();
+}
+}
+
+int main(int argc, char* argv[]) try {
+    if (argc != 5) {
+        throw std::runtime_error(std::string{"Usage: "} + argv[0]
+            + " <openvino_model.xml> <tokenizer.xml> <detokenizer.xml> '<prompt>'");
```

Comment:
Can we simplify command line for model passing: expect one parameter with model directory with all three IRs inside and hardcode names of those IRs in the code? It is somewhat we would expect from future development of optimum-intel where both tokenizer and detokenizer will be exported alongside with the model. We already have a name for the main model: `openvino_model.xml` and we can introduce `openvino_tokenizer.xml` and `openvino_detokenizer.xml` correspondingly. Now they are called `tokenizer.xml` and `detokenizer.xml` but better to align with optimum-intel naming for the main model.

Please collect this as a potential improvement.

Reply:
Done